projects/igniteui-angular/src/lib/data-operations/data-util.ts

```diff
@@ -152,7 +153,7 @@ export class DataUtil {
             const rowId = primaryKey ? item[primaryKey] : item;
             const transaction = transactions.find(t => t.id === rowId);
             if (transaction && transaction.type === TransactionType.UPDATE) {
-                data[index] = transaction.newValue;
+                data[index] = mergeObjects(data[index], transaction.newValue);
```

Comment:
`transactions` parameter here is passed from `grid.transactions.getAggregatedChanges(true)`. In this way we are always receive in `transaction.newValue` cloned value of the row data. No need to merge it in the `data[index]`. Doing so updates the original data which should not happen. Either do not merge `newValue` in row data (`data[index]`), or clone the row data using provided clone strategy.